arkouda/categorical.py

```diff
@@ -595,7 +596,8 @@ def register(self, user_defined_name:str) -> Categorical:
         Objects registered with the server are immune to deletion until
         they are unregistered.
         """
-        [getattr(self, p).register(f"{user_defined_name}.{p}") for p in Categorical.RegisterablePieces]
+        [getattr(self, p).register(f"{user_defined_name}.{p}") for p in Categorical.RegisterablePieces
+         if p in Categorical.RequiredPieces or getattr(self, p) is not None]
```

Comment:
I recommend putting this logic within a python closure method and adding some pydoc comments